terraform/transform_transitive_reduction.go

```diff
@@ -0,0 +1,20 @@
+package terraform
+
+// TransitiveReductionTransformer is a GraphTransformer that performs
+// finds the transitive reduction of the graph. For a definition of
+// transitive reduction, see Wikipedia.
```

Comment:
Yes, I figured if you don't know what a transitive reduction is, then saying "maximally reduces the edges in a graph to ensure the same reachability" probably still wouldn't make sense. ;)